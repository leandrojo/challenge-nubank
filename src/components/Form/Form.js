import styled from 'styled-components';

export function styleProps(props) {
  return props.css;
}

const Form = styled.form`
  display: flex;
  flex: 1;
  flex-direction: column;

  ${styleProps};
`;

Form.defaultProps = {
  css: {},
};

export default Form;
