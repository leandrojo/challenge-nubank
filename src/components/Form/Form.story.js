/* eslint-disable import/no-unresolved */

import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';

import { Form, Input } from 'components';

const ProviderDecorator = storyFn => <div>{storyFn()}</div>;

storiesOf('form', module)
  .addDecorator(ProviderDecorator)
  .add('advanced validation (email)', () => {
    const FormSample = () => {
      const onSubmit = () => action('submit');

      return (
        <Form autoComplete="off" onSubmit={onSubmit} initialValues={{}}>
          <Input
            label="E-mail"
            name="email"
            rules={[
              {
                type: 'string',
                required: true,
                message: 'Campo obrigat칩rio.',
              },
              {
                validator(rule, value, callback, source) {
                  const re = /\S+@\S+\.\S+/;
                  const errors = [];
                  if (re.test(source.email) === false) {
                    errors.push(new Error('Entre com um e-mail v치lido.').message);
                  }
                  callback(errors);
                },
              },
            ]}
          />
        </Form>
      );
    };

    return <FormSample />;
  })
  .add('rules when submit', () => {
    const FormSample = () => {
      const onSubmit = () => action('submit');

      return (
        <Form autoComplete="off" onSubmit={onSubmit} initialValues={{}}>
          <Input
            label="Nome Completo"
            name="name"
            rules={[
              {
                type: 'string',
                required: true,
                message: 'Campo obrigat칩rio.',
              },
            ]}
          />

          <Input
            label="Data de Nascimento"
            name="birthDate"
            pattern="99/99/9999"
            rules={[
              {
                type: 'string',
                required: true,
                message: 'Campo obrigat칩rio.',
              },
            ]}
          />
        </Form>
      );
    };

    return <FormSample />;
  });
