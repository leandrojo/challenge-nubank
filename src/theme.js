const fontFamily = 'Rubik, sans-serif';

const theme = {
  backgroundColor: '#f5f5f5',
  colors: {
    primary: '#7f39b7',
    grayExtraDark: '#555',
    grayDark: '#767676',
    gray: '#999',
    grayLight: '#ccc',
    grayExtraLight: '#FAFAFA',
  },
  fontFamily,
  fontSize: '1em',
  typography: {
    heading: {
      color: '#252226',
    },
    paragraph: {
      color: '#716d73',
    },
  },
};

theme.components = {
  input: {
    css: {
      backgroundColor: 'transparent',
      fontFamily,
      fontSize: '1.4em',
      fontWeight: 300,
    },
  },
  pane: {
    backgroundColor: '#fff',
  },
  range: {
    emptyColor: theme.colors.grayLight,
  },
};

export default theme;
