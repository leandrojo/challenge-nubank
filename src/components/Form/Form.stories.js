/* eslint-disable import/no-extraneous-dependencies */
/* eslint-disable max-len */

import React from 'react';
import styled from 'styled-components';

import { Input as Component } from '..';
import Form from './Form';

const Input = styled(Component)`
  border-bottom: 2px solid black;
`;

export default {
  title: 'components/Form',
};

export const custom = () => (
  <Form
    css={`
      background: #fff;
      display: block;
      padding: 20px;

      p:first-of-type {
        margin: 20px 0 0;
      }
    `}
  >
    <Input placeholder="This is a Input text." />
    <p>Form is like Box component, a simple styled component customize. Here is custom background and padding.</p>
    <p>Use only as wrapper.</p>
  </Form>
);

custom.story = {
  name: 'custom',
};
