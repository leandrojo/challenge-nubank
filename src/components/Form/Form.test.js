/* eslint-env jest */

import React, { useState } from 'react';
import { ThemeProvider } from 'styled-components';
import renderer from 'react-test-renderer';

import theme from 'theme';

import { Input } from '..';
import Form from './Form';

describe('Form', () => {
  it('with initial value', () => {
    const Component = () => {
      const [value, onChange] = useState(1000);

      return (
        <ThemeProvider theme={theme}>
          <Form>
            <Input name="firstDeposit" onChange={onChange} value={value} />
          </Form>
        </ThemeProvider>
      );
    };

    const expected = renderer.create(<Component />).toJSON();

    expect(expected).toMatchSnapshot();
  });
});
