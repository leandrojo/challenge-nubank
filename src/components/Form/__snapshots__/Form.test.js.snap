// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form with initial value 1`] = `
<form
  className="sc-htpNat flIVyN"
>
  <input
    className="sc-bxivhb dRyoeN"
    id=""
    name="firstDeposit"
    onBlur={[Function]}
    onChange={[Function]}
    onError={[Function]}
    onFocus={[Function]}
    pattern="\\\\d*"
    type="text"
    value=""
  />
</form>
`;
